backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java

```diff
@@ -1067,6 +1085,24 @@ private static boolean proxyEnabled() {
         return Config.<Boolean>getValue(ConfigValues.ImageTransferProxyEnabled);
     }
 
+    @Override
+    protected VDS checkForActiveVds() {
+        List<VDS> hosts = getAllRunningVdssInPool()
+                .stream()
+                .filter(vds -> resourceManager.getVdsManager(vds.getId())
+                        .getDomains()
+                        .stream()
+                        .filter(vdsDomainsData -> vdsDomainsData.getDomainId().equals(getStorageDomainId()))
+                        .allMatch(VDSDomainsData::isValid))
+                .collect(Collectors.toList());
+        if (!hosts.isEmpty()) {
+            return hosts.get(new Random().nextInt(hosts.size()));
+        }
```

Comment:
I saw a nice code to get a random element from a stream:
```strings
            .stream()
            .sorted((o1, o2) -> ThreadLocalRandom.current().nextInt(-1, 2))
            .findAny()
            .get();
```
something like that can be nice so we won't need to analyze all the hosts - but as the list of hosts is not that long, that's just a nice to have

Reply:
ah but you actually don't do an heavy call to check each domain so it's probably more or less like sorting..

Reply:
I copied this from the parent and only added the filter, I think using the helper in VdsCommandsHelper would be more pleasant 
https://github.com/oVirt/ovirt-engine/blob/1c3f4ea9ab33b94a237041648a774fd57b83abe8/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/utils/VdsCommandsHelper.java#L153-L163